soccerdata/fbref.py

```diff
@@ -1167,15 +1135,19 @@ def _concat(dfs: List[pd.DataFrame], key: List[str]) -> pd.DataFrame:
         columns.loc[mask, 1] = ""
         column_idx = pd.MultiIndex.from_tuples(columns.to_records(index=False).tolist())
 
-        for df in dfs:
+        for i, df in enumerate(dfs):
             if df.columns.equals(column_idx):
                 # This dataframe already has the uniform column index
                 pass
-            elif len(df.columns) == len(column_idx):
+            if len(df.columns) == len(column_idx):
                 # This dataframe has the same number of columns and the same
                 # level 1 columns, we assume that the level 0 columns can be
                 # replaced
                 df.columns = column_idx
+            else:
+                # This dataframe has a different number of columns, so we want
+                # to make sure its columns match with column_idx
+                dfs[i] = df.reindex(columns=column_idx, fill_value=None)
```

Comment:
Note that this only works if the forfeited game did not take place before all other games that are scraped. For example, the code below will return a data frame with only the columns that are present for the forfeited one.

```py
import soccerdata as sd
fbref_seriea = sd.FBref(["ITA-Serie A"], 2021)
fbref_seriea.read_player_match_stats(match_id=["e0a20cfe", "a3e10e13"])
```

You could use some heuristics to determine the most likely correct combination (such as determining which combination of columns is the most frequent one) or check whether the scoreline has a "*". 

Actually I don't care too much about this edge case, but if you do not plan to address it, could you at least document it?

Reply:
I changed `test_concat_with_forfeited_game` to test for this case. It looks like the order of matches doesn't matter, you'd always get the same result.

Reply:
Indeed, the order in the `match_id` list does not matter. The matches are sorted by their date. Note that "a3e10e13" is a match that was played after the forfeited "e0a20cfe" match.

Reply:
I updated the test accordingly. I also changed `_concat` to use the columns from the dataframe with the most columns.